### PR TITLE
Use POST /parse instead of GET /parse to submit programs.

### PR DESCRIPTION
**Pull request by fpereiro**
Currently we use a GET request to submit a program from the client to the server for parsing (`GET /parse`). GET requests are recommended to not exceed 2048 characters, because of limited support for long URLs on some browsers (such as IE11).

By switching to POST, we can sidestep this problem. It is also somewhat more RESTful to use POST rather than GET for this type of operation, since we're in effect submitting data.

This SO answer has further info on this subject: https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers

### CODE REVIEW

**Comment by Felienne** [comment]
Hi @fpereiro!

Can you rebase this one or otherwise mix the merge issues? I want to test this locally extensively before merging because it might impact everything.

Felienne

**Comment by fpereiro** [comment]
@Felienne  Just merged the latest master onto the branch; there should be no conflicts now :)

File: app.py [hunk 1]
```diff
@@ -72,11 +72,18 @@
...
 
-@app.route('/parse/', methods=['GET'])
+@app.route('/parse', methods=['POST'])
 def parse():
-    # Retrieve the name from url parameter
-    code = request.args.get("code", None)
-    level = int(request.args.get("level", None))
+    body = request.json
+    if not body:
+        return "body must be an object", 400
```
Comment (rix0rrr) [inline app.py:79]: This looks weird. Are you sure this is the way to return a 400 response?

Comment (fpereiro) [reply app.py:79]: Good question. I just checked and it does seem to return the string as the body and the integer as the response code. I found this line in this official doc (http://www.flaskapi.org/api-guide/status-codes/): `return content, status.HTTP_404_NOT_FOUND`

**Pull request merged**